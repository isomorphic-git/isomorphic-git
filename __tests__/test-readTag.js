/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// @ts-ignore
const snapshots = require('./__snapshots__/test-readTag.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')

const { readTag } = require('isomorphic-git')

describe('readTag', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('annotated tag', async () => {
    // Setup
    const { gitdir } = await makeFixture('test-readTag')
    // Test
    const tag = await readTag({
      gitdir,
      oid: '587d3f8290b513e2ee85ecd317e6efecd545aee6'
    })
    expect(tag).toMatchSnapshot()
  })
})
