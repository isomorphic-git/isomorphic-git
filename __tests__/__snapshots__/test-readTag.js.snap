// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readTag annotated tag 1`] = `
Object {
  "oid": "587d3f8290b513e2ee85ecd317e6efecd545aee6",
  "payload": "object 033417ae18b174f078f2f44232cb7a374f4c60ce
type commit
tag mytag
tagger William Hilton <wmhilton@gmail.com> 1578802395 -0500

This is a tag message.

",
  "tag": Object {
    "message": "This is a tag message.
",
    "object": "033417ae18b174f078f2f44232cb7a374f4c60ce",
    "signature": undefined,
    "tag": "mytag",
    "tagger": Object {
      "email": "wmhilton@gmail.com",
      "name": "William Hilton",
      "timestamp": 1578802395,
      "timezoneOffset": 300,
    },
    "type": "commit",
  },
}
`;
